### PR TITLE
Update button mouseover when switching menus.

### LINKED ISSUE #5594: UI elements don't display proper state on creation
Example, from the main menu click Extras. The Credits button will appear under the mouse. Until the mouse is moved the state of the button will be as if in a non-mouseover state.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/2580435-ui-elements-don-t-display-proper-state-on-creation?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pchote**
Fixes #5594.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
I don't recall experiencing the original issue but sure :+1: menu switching works fine.

**Comment by pchote** [comment]
The issue is that you were left with your mouse over a button, but that button did not show its mouseover state.

File: OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs [hunk 1]
```diff
@@ -165,29 +173,29 @@ public MainMenuLogic(Widget widget, World world)
...
 			loadMapButton.OnClick = () =>
 			{
-				menuType = MenuType.None;
+				SwitchMenu(MenuType.None);
 				Game.OpenWindow("MAPCHOOSER_PANEL", new WidgetArgs()
 				{
 					{ "initialMap", null },
 					{ "initialTab", MapClassification.User },
-					{ "onExit", () => menuType = MenuType.MapEditor },
+						{ "onExit", () => SwitchMenu(MenuType.MapEditor) },
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs]: Why the extra tab here?

**Comment by pchote** [comment]
Fixed and rebased.

**Comment by Mailaender** [comment]
To reproduce: press the left mouse button and move the cursor without releasing it. This doesn't seem to fix it though.

**Comment by Mailaender** [comment]
No wait, that is a different bug. This is about pressing a button that opens a sub-menu and you need to move the mouse to trigger the highlight. It will now immediately get highlighted. :white_check_mark:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ad54d7f64b9a42aaaf91501b086b8d94588afc31...f7f802d54ee66ca577fa69173c46bc9fd3549424)

**Pull request merged**